core/src/main/java/org/mule/runtime/core/internal/util/rx/Operators.java

```diff
@@ -30,13 +29,13 @@ private Operators() {}
    * @return custom operator {@link BiConsumer} to be used with {@link reactor.core.publisher.Flux#handle(BiConsumer)}.
    */
   public static BiConsumer<Event, SynchronousSink<Event>> nullSafeMap(Function<Event, Event> mapper) {
-    return (t, sink) -> {
-      if (t != null) {
-        Event r = mapper.apply(t);
-        if (r != null) {
-          sink.next(r);
+    return (event, sink) -> {
+      if (event != null) {
+        Event result = mapper.apply(event);
+        if (result != null) {
+          sink.next(result);
         } else {
-          sink.error(newEventDroppedException(t));
+          event.getContext().success();
```

Comment:
This is exactly what this PR is doing:
Removing the use of an exception to represent a dropped message, which then handled at the end of the flow with ` event.getContext().success(); to instead complete with no response immediately.  The reason I used exception in the first place was to handle the enricher/response-adaptor scenarios where we need to handle the null rather than respond null, but I'm now using a different approach to do this using a child context and can therefore clean out all use of ExceptionDroppedException.